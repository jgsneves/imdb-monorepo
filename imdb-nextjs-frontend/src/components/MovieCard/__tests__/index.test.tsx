import { MovieCard } from "..";
import { render } from "@testing-library/react";

jest.mock("next/router", () => {
  const originalModule = jest.requireActual("next/router");
  return {
    __esModule: true,
    ...originalModule,
    useRouter: jest.fn(),
  };
});

describe("MovieCard test suite", () => {
  it("should render proper snapshot", () => {
    const { asFragment } = render(
      <MovieCard
        actors={["José", "João", "Maria"]}
        director="Nome do diretor"
        genre="Ação"
        id="id"
        name="Nome do filme"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
