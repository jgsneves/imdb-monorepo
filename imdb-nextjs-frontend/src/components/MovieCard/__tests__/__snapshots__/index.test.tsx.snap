// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard test suite should render proper snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-card css-r29fgq"
  >
    <img
      alt="Caffe Latte"
      class="chakra-image css-1nra699"
      src="https://images.unsplash.com/photo-1667489022797-ab608913feeb?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxlZGl0b3JpYWwtZmVlZHw5fHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60"
    />
    <div
      class="chakra-stack css-8g8ihq"
    >
      <div
        class="chakra-card__body css-0"
      >
        <h2
          class="chakra-heading css-0"
        >
          Nome do filme
        </h2>
        <p
          class="chakra-text css-o5g2qp"
        >
          Diretor: Nome do diretor
        </p>
        <p
          class="chakra-text css-o5g2qp"
        >
          Gênero: Ação
        </p>
        <p
          class="chakra-text css-o5g2qp"
        >
          Atores: José, João, Maria
        </p>
      </div>
      <div
        class="chakra-card__footer css-k008qs"
      >
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          ver detalhes
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
